client/src/main/scala/org/apache/celeborn/client/commit/ReducePartitionCommitHandler.scala

```diff
@@ -75,6 +75,7 @@ class ReducePartitionCommitHandler(
   private val stageEndShuffleSet = ConcurrentHashMap.newKeySet[Int]()
   private val inProcessStageEndShuffleSet = ConcurrentHashMap.newKeySet[Int]()
   private val shuffleMapperAttempts = JavaUtils.newConcurrentHashMap[Int, Array[Int]]()
+  private val shuffleToCompletedMappers = JavaUtils.newConcurrentHashMap[Int, Int]()
```

Comment:
millions of key-value pairs take dozens of MiB memory, not sure if worth using `fastutil`-like optimization, which is more memory-friendly for primitive types.

Reply:
Sounds good - let me add a note.